test/enums.jl

```diff
@@ -167,3 +169,34 @@ end
 @test Int(haggis) == 4
 
 @test (Vector{Fruit}(undef, 3) .= apple) == [apple, apple, apple]
+
+# long, discongruous
+@enum Alphabet begin
+    alphabet_a = 27
```

Comment:
I'm not sure this will hit the intended line, since the values are still continuous (`length(values) == hi - lo + 1`).

Reply:
Ok that makes sense, I bumped it up to 28

Reply:
That doesn't change anything; the test `length(values) == hi - lo + 1` will still be true.

Reply:
Ok sorry. How should I change it? I was thinking that alphabet_b would be assigned to 1, so that (27 - 1 + 1) > 26

Reply:
Ok sorry. How should I change it? I was thinking that alphabet_b would be assigned to 1, so that `(27 - 1 + 1) > 26`

Reply:
Ah, I see. The default value is the previous value plus one:
```
julia> @enum X a=10 b c=20 d

julia> a
a::X = 10

julia> b
b::X = 11

julia> c
c::X = 20

julia> d
d::X = 21
```

Reply:
Ok, I think this new version will work based on the output of `@macroexpand`